internal/config/config.go

```diff
@@ -517,6 +533,45 @@ func (c *Config) getFieldIDs(client *jira.Client) (*fields, error) {
 	return &fieldIDs, nil
 }
 
+// getComponents resolves every component set in config against
+// Jira project, and saves these components used by issue-sync.
+func (c *Config) getComponents(proj *jira.Project) ([]*jira.Component, error) {
+	var returnComponents []*jira.Component
+
+	componentsStr := c.cmdConfig.GetString(options.ConfigKeyJiraComponents)
+
+	if componentsStr == "" {
+		return returnComponents, nil
+	}
+
+	components := strings.Split(componentsStr, ",")
+
+	for _, configComponent := range components {
+		found := false
+
+		for j := range proj.Components {
+			projComponent := &proj.Components[j]
+
+			if projComponent.Name == configComponent {
+				found = true
+				foundComponent := jira.Component{
+					Name: projComponent.Name,
+					ID:   projComponent.ID,
+				}
+
+				returnComponents = append(returnComponents, &foundComponent)
```

Comment:
Instead of append I would define a fix size slice because we know the possible maximum element number in the slice and at the end of the function I would cut the unused part of the slice.

If you wouldn't like to use a fixed size memory array, you can define the capacity of the slice and in this case you also can use the append function.

Reply:
Again, using the pattern found in case of issues, labels.